### PR TITLE
Refactor to use a single Asset model

### PR DESCRIPTION
**Pull request by StevenClontz**
Reduce business complexity by refactoring to use a single Asset model. Summary:

## Summary
Collapsed the two-model, many-to-many asset design (`LibraryAsset` user-owned + `ProjectAsset` join row) into a single `Asset` model that `belongs_to :project` directly. All asset writes now go exclusively through `Project#assets_attributes=` (nested attributes) — no dedicated `AssetsController` create/update/destroy actions.

## Data migration
Three-step migration (`db/migrate/20260707120000..2_*`):
1. **Create `assets` table** — `project_id`, `ref`, `kind`, `source`, `short_description`, `description`, `title`.
2. **Migrate data** from `library_assets`/`project_assets`, handling three cases:
   - **1 project membership** (common case): reuse the `LibraryAsset`'s id as the new `Asset.id`; flip the ActiveStorage attachment's `record_type` from `"LibraryAsset"` → `"Asset"` (no re-upload).
   - **Multiple project memberships** (legal under the old many-to-many): first membership reuses the id; every extra membership gets a **new** `Asset` row + a new attachment row sharing the same `blob_id`.
   - **Zero memberships (orphans)**: grouped into **one new "Imported Assets" project per user**, with a root division that references each recovered asset via `<plus:image ref="..."/>`.
3. **Drop** `library_assets` and `project_assets` tables/FKs.

Verified against real dev data (not just fixtures): 13 assets, 3 new holding projects created correctly, all file attachments still resolve.

## Backend changes
| Area | Change |
|---|---|
| `app/models/asset.rb` (new) | `belongs_to :project`, `has_one_attached :file`, `kind` enum, `url` (ported verbatim incl. SVG inline-disposition security override), `ref` uniqueness/format validation + cross-check vs `Division`. **No** `set_to_correct_user` — structurally impossible now since an Asset can't be shared across projects. |
| `app/models/project.rb` | `has_many :assets, dependent: :destroy` + `assets_attributes=` (client-minted-UUID pattern, same as `divisions_attributes=`). `full_dup` now copies each Asset's own attributes + re-attaches the same file blob — gives every duplicate (same- or cross-owner) an independent row. |
| `app/models/division.rb` | `ref_unique_among_project_assets` → `ref_unique_among_assets`, now checks `Asset`. |
| `app/models/user.rb` | `has_many :library_assets` → `has_many :assets, through: :projects`. |
| `app/models/ability.rb` | Single `can :manage, Asset, project: { user_id: user.id }` replacing the separate `ProjectAsset`/`LibraryAsset` rules. |
| Controllers | Deleted `LibraryAssetsController` + `ProjectAssetsController`. New `AssetsController` with just two read-only redirects: `share` (public, project+ref-scoped — used by editor preview and published builds) and `file` (deprecated legacy owner-only id-scoped link). `ProjectsController#project_params` now permits `assets_attributes` (including `:file` for multipart uploads). |
| `app/jobs/fetch_build_zip_job.rb` | Reads `project.assets` directly instead of `project_assets` → `library_asset`. |
| Routes | Removed `/library`, `/library/:id/file`, `/projects/:id/project_assets`; `project_assets#share` → `assets#share` (same path/helper); legacy `share_assets/external/:id` repointed at `assets#file`. |
| Views | `app/views/assets/_asset.json.jbuilder` replaces the `library_assets`/`project_assets` partials; embedded in `projects/_project.json.jbuilder` as `project.assets`. |

## Tests
- `test/fixtures/assets.yml` replaces `library_assets.yml` + `project_assets.yml`.
- `test/models/asset_test.rb` replaces `library_asset_test.rb` + `project_asset_test.rb` (url behavior, ref-vs-division uniqueness, and a new `full_dup` independence test).
- `test/controllers/assets_controller_test.rb` replaces the old controller tests.
- Updated `division_test.rb`, `project_test.rb` (`assets_attributes`), `projects_controller_test.rb` (added a real multipart upload test; rewrote the `full_dup` copy-independence test), `fetch_build_zip_job_test.rb`, `asset_fetches_controller_test.rb`.
- **168 tests, 0 failures.**

## Explicit behavior changes (intended, flagged for sign-off)
- Removing an asset from a project is now a **real delete** — no more "keep in personal library" since cross-project reuse no longer exists.
- `full_dup` no longer shares a row between original and copy even for same-owner copies (previously a latent bug: editing/deleting in one project silently affected the other).

## Not done (explicitly out of scope this pass)
`app/javascript/controllers/react/editor.jsx` and `tryit.jsx` were **not** refactored — they still call the old `/library`, `/library/:id`, and `/projects/:id/project_assets` routes with the old nested JSON shape, so the live editor's asset-management UI will break against this backend until a follow-up frontend pass lands.

### CODE REVIEW

File: app/views/assets/_asset.json.jbuilder [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  # `file`: the public, project+ref-scoped redirect to the asset's current
+  # file location -- a real, directly fetchable URL, used both as the
+  # editor's own thumbnail `<img src>` and by the client when it needs to
+  # fetch the bytes.
+  # `extension`: lets the client build the *bare* `<ref>.<ext>` fragment it
+  # writes into the assembled PreTeXt as an image's `source` -- the build
+  # server treats that as a plain external-asset filename and prepends
+  # `external/` itself, so embedding a full URL there double-prefixes (see
+  # the comments in projects_controller.rb / project.rb).
+  json.file share_asset_project_path(asset.project, ref: asset.ref, format: extension)
```
Comment (StevenClontz) [inline app/views/assets/_asset.json.jbuilder]: rename to `path` since it's a string with a path, not an actual file?

Comment (StevenClontz) [reply app/views/assets/_asset.json.jbuilder]:
```suggestion
  json.path share_asset_project_path(asset.project, ref: asset.ref, format: extension)
```

**Comment by StevenClontz** [comment]
As mentioned to @oscarlevin this is now ready to refactor with the editor. Idea: editor state can match the shape of the projects rails API:

```js
{
  "attribute": value, ...
  divisions: [
    { "attribute": value, ... }, ...
  ],
  assets: [
    { "attribute": value, ... }, ...
  ]
}
```

See https://api.rubyonrails.org/v8.0/classes/ActiveRecord/NestedAttributes/ClassMethods.html (though this doesn't get into the business of the shape of the query params, something like `project[divisions_attributes][id]=123-456-789`).

**Comment by oscarlevin** [comment]
I've implemented the javascript parts of this.  Should be ready to go.  Claude found an edge case for divisions that were added and then immediately removed (second to last commit).  Otherwise I think it is all js and tests.

**Comment by oscarlevin** [comment]
@StevenClontz anything I can do to help move this along?

**Comment by StevenClontz** [comment]
No -- I'll review and hopefully merge on Saturday but if you feel bold you can try to merge earlier.

File: db/schema.rb [hunk 2]
```diff
@@ -68,14 +68,17 @@
   create_table "build_files", id: :uuid, default: -> { "gen_random_uuid()" }, force: :cascade do |t|
     t.uuid "build_id", null: false
     t.datetime "created_at", null: false
-    t.string "relative_path"
+    t.string "relative_path", null: false
     t.datetime "updated_at", null: false
+    t.index ["build_id", "relative_path"], name: "index_build_files_on_build_id_and_relative_path", unique: true
```
Comment (StevenClontz) [inline db/schema.rb]: Watch out for this gotcha @oscarlevin -- when juggling branches that touch the database, be sure to rollback migrations to match the main branch. Otherwise `schema.rb` will infer incorrect columns/migrations by reading your database.

**Comment by StevenClontz** [comment]
This is ready to merge, except that it will disrupt active users. We should merge and run the migration at a less disruptive time.

**Comment by oscarlevin** [comment]
Yay!  I'm happy to trigger the merge late tonight when we have no active
users.

**Comment by StevenClontz** [comment]
Insomnia struck so I pushed the button. Looks like it deployed with no issues to report.
